plugins/baseplugin/src/main/java/org/apache/unomi/plugins/baseplugin/conditions/accessors/HardcodedPropertyAccessorRegistry.java

```diff
@@ -0,0 +1,131 @@
+        propertyAccessors.put(Campaign.class.getName(), new CampaignHardcodedPropertyAccessor(this));
+        propertyAccessors.put(Map.class.getName(), new MapHardcodedPropertyAccessor(this));
+    }
+
+    public static class NextTokens {
+        public String propertyName;
+        public String leftoverExpression;
+    }
+
+    protected NextTokens getNextTokens(String expression) {
+        NextTokens nextTokens = new NextTokens();
+        if (expression.startsWith("[\"")) {
+            int lookupNameBeginPos = "[\"".length();
+            int lookupNameEndPos = expression.indexOf("\"].", lookupNameBeginPos);
+            if (lookupNameEndPos > lookupNameBeginPos) {
+                nextTokens.propertyName = expression.substring(lookupNameBeginPos, lookupNameEndPos);
+                nextTokens.leftoverExpression = expression.substring(lookupNameEndPos+2);
+            } else {
+                nextTokens.propertyName = expression.substring(lookupNameBeginPos);
+                nextTokens.leftoverExpression = null;
+            }
+        } else if (expression.startsWith(".")) {
+            int lookupNameBeginPos = ".".length();
+            int lookupNameEndPos = lookupNameBeginPos;
+            int dotlookupNameEndPos = expression.indexOf(".", lookupNameBeginPos);
+            int squareBracketlookupNameEndPos = expression.indexOf("[", lookupNameBeginPos);
+            if (dotlookupNameEndPos > lookupNameBeginPos && squareBracketlookupNameEndPos > lookupNameBeginPos) {
+                lookupNameEndPos = Math.min(dotlookupNameEndPos, squareBracketlookupNameEndPos);
+            } else if (dotlookupNameEndPos > lookupNameBeginPos) {
+                lookupNameEndPos = dotlookupNameEndPos;
+            } else if (squareBracketlookupNameEndPos > lookupNameBeginPos) {
+                lookupNameEndPos = squareBracketlookupNameEndPos;
+            } else {
+                lookupNameEndPos = -1;
+            }
+            if (lookupNameEndPos > lookupNameBeginPos) {
+                nextTokens.propertyName = expression.substring(lookupNameBeginPos, lookupNameEndPos);
+                nextTokens.leftoverExpression = expression.substring(lookupNameEndPos);
+            } else {
+                nextTokens.propertyName = expression.substring(lookupNameBeginPos);
+                nextTokens.leftoverExpression = null;
+            }
+        } else {
+            int lookupNameBeginPos = 0;
+            int lookupNameEndPos = expression.indexOf(".", lookupNameBeginPos);
+            if (lookupNameEndPos > lookupNameBeginPos) {
+                nextTokens.propertyName = expression.substring(lookupNameBeginPos, lookupNameEndPos);
+                nextTokens.leftoverExpression = expression.substring(lookupNameEndPos);
+            } else {
+                nextTokens.propertyName = expression.substring(lookupNameBeginPos);
+                nextTokens.leftoverExpression = null;
+            }
```

Comment:
this code is duplicated in the 3 cases, we could move it out of the tree cases and just calculated the `lookupNameBeginPos, lookupNameEndPos.`
Then do the this code at the end of the 3 cases check.